src/resources/js/routes/Administration.svelte

```diff
@@ -0,0 +1,328 @@
+  // errors          Array of strings for meetings for which the server returned an error code.  The string should consist of the bmlt_id and then the error returned by the server.
+  interface Stats {
+    totalRows: number;
+    malformedRows: number[];
+    updated: number[];
+    noUpdateNeeded: number[];
+    notedAsDeleted: number[];
+    notFound: number[];
+    errors: string[];
+  }
+  let stats: Stats = $state({
+    totalRows: 0,
+    malformedRows: [],
+    updated: [],
+    noUpdateNeeded: [],
+    notedAsDeleted: [],
+    notFound: [],
+    errors: []
+  });
+
+  async function processFile(file: File): Promise<void> {
+    try {
+      isLoading = true;
+      errorMessage = null;
+      // Reset stats
+      stats = {
+        totalRows: 0,
+        malformedRows: [],
+        updated: [],
+        noUpdateNeeded: [],
+        notedAsDeleted: [],
+        notFound: [],
+        errors: []
+      };
+
+      console.log('Processing file:', file.name, file.type);
+
+      const extension = getFileExtension(file.name);
+
+      if (!isValidFileType(extension)) {
+        throw new Error('Unsupported file type. Please upload a CSV or Excel file.');
+      }
+
+      const XLSX = await import('xlsx');
+      const buffer = await file.arrayBuffer();
+      const workbook = XLSX.read(buffer, {
+        type: 'array',
+        raw: extension === 'csv'
+      });
+
+      const firstSheetName = workbook.SheetNames[0];
+      const worksheet = workbook.Sheets[firstSheetName];
+      // Convert the spreadsheet to json using the { header: 1 } option so that it explicitly includes the header row.  Then we can check that
+      // the header contains 'bmlt_id' and 'Committee'.  If we didn't use the option, we couldn't tell whether a spreadsheet was missing the
+      // worldId column altogether or if the heading was there but the worldId values were the empty string (which is legal).
+      const jsonData = XLSX.utils.sheet_to_json(worksheet, { header: 1 });
+      if (jsonData.length < 2) {
+        throw new Error('The file has no readable data.');
+      }
+      const header = jsonData[0] as string[];
+      const bmltIdIndex = header.findIndex((c) => c === 'bmlt_id');
+      const worldIdIndex = header.findIndex((c) => c === 'Committee');
+      if (bmltIdIndex < 0 || worldIdIndex < 0) {
+        throw new Error('The file is missing a column for bmlt_id or Committee or both.');
+      }
+
+      stats.totalRows = jsonData.length - 1;
+
+      // Build a map of meetingId -> worldId from the CSV
+      const updateMap = new Map<number, string>();
+      const meetingIds: number[] = [];
+
+      for (let i = 1; i < jsonData.length; i++) {
+        const row = jsonData[i] as any[];
+        const meetingId = row[bmltIdIndex];
+        if (typeof meetingId !== 'number') {
```

Comment:
this will fail currently for CSVs, should take suggestion above or attempt to covert string to number explicitly